src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -79,12 +74,12 @@ def grid(gridspec):
 
     Parameters
     ----------
-    gridspec : dict
+    gridspec : dict[str, list[str]]
         A mapping from parameter names to lists of parameter values.
 
     Yields
     ------
-    Iterable[dict]
+    Generator[dict[str, str]]
```

Comment:
I think the "Yields" section implies that this is a generator already. This yields dictionaries, but the docs imply that it yields generators of dictionaries.

https://numpydoc.readthedocs.io/en/latest/format.html#yields

As a note, the return type above of `-> Generator[...]` should be fine as-is.
```suggestion
    dict[str, str]
```

Reply:
Updated